src/input/ButtonInputAdapter.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "InputInterface.h"
+#include <Button.h>
+
+/**
+ * @class ButtonInputAdapter
+ * @brief Adapter class to handle button input for an LCD menu.
+ *
+ * This class implements the InputInterface to provide button input handling
+ * for an LCD menu. It observes button presses and processes commands accordingly.
+ *
+ * @details
+ * The ButtonInputAdapter class encapsulates a Button object and a command byte.
+ * It initializes the button and observes its state to trigger menu commands when
+ * the button is pressed.
+ *
+ * @param menu Pointer to the LcdMenu object that this adapter will interact with.
+ * @param button The Button object that will be used for input.
+ * @param command The command byte that will be sent to the menu when the button is pressed.
+ */
+class ButtonInputAdapter : public InputInterface {
```

Comment:
If it wraps `Button`, then it should be named `ButtonAdapter`?

Reply:
Done in different PR